sound/soc/sof/loader.c

```diff
@@ -315,75 +347,91 @@ static void sof_get_windows(struct snd_sof_dev *sdev)
 		return;
 	}
 
-	for (i = 0; i < sdev->info_window->num_windows; i++) {
-		elem = &sdev->info_window->window[i];
+	count = sdev->xman_info_window ?
+		sdev->xman_info_window->num_windows :
+		sdev->info_window->num_windows;
+
+	for (i = 0; i < count; i++) {
+		if (sdev->xman_info_window) {
+			xelem = &sdev->xman_info_window->window[i];
+			elem_id = xelem->id;
+			elem_type = xelem->type;
+			elem_size = xelem->size;
+			elem_offset = xelem->offset;
+		} else {
+			elem = &sdev->info_window->window[i];
+			elem_id = elem->id;
+			elem_type = elem->type;
+			elem_size = elem->size;
+			elem_offset = elem->offset;
+		}
 
-		window_offset = snd_sof_dsp_get_window_offset(sdev, elem->id);
+		window_offset = snd_sof_dsp_get_window_offset(sdev, elem_id);
 		if (window_offset < 0) {
 			dev_warn(sdev->dev, "warn: no offset for window %d\n",
-				 elem->id);
+				 elem_id);
 			continue;
 		}
 
-		switch (elem->type) {
+		switch (elem_type) {
 		case SOF_IPC_REGION_UPBOX:
-			inbox_offset = window_offset + elem->offset;
-			inbox_size = elem->size;
+			inbox_offset = window_offset + elem_offset;
+			inbox_size = elem_size;
 			snd_sof_debugfs_io_item(sdev,
 						sdev->bar[bar] +
 						inbox_offset,
-						elem->size, "inbox",
+						elem_size, "inbox",
 						SOF_DEBUGFS_ACCESS_D0_ONLY);
 			break;
 		case SOF_IPC_REGION_DOWNBOX:
-			outbox_offset = window_offset + elem->offset;
-			outbox_size = elem->size;
+			outbox_offset = window_offset + elem_offset;
+			outbox_size = elem_size;
```

Comment:
do all those mechanical changes in a separate patch first, then add the real changes in a follow-up

Reply:
done